src/live_restore/live_restore_fs.c

```diff
@@ -330,14 +341,11 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
              * list.
              */
             bool add_source_file = false;
-            if (WT_SUFFIX_MATCH(dirlist_src[i], WTI_LIVE_RESTORE_STOP_FILE_SUFFIX)) {
-                /*
-                 * FIXME-WT-14040: It is possible for stop files to exist in the source directory.
-                 * Although those files are cleaned up on completion the concurrency management was
-                 * not implemented in that change so we can't guarantee they don't exist.
-                 */
-                continue;
-            }
+            /*
+             * Stop files should never exist in the source directory. We check this on startup but
+             * add a sanity check here.
+             */
+            WT_ASSERT(session, !WT_SUFFIX_MATCH(dirlist_src[i], WTI_LIVE_RESTORE_STOP_FILE_SUFFIX));
```

Comment:
Make it an `_ALWAYS`.

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11590/commits/e7accc327ae230d77e871da8c444dbdda9561757